scope/xgb.py

```diff
@@ -0,0 +1,129 @@
+import datetime
+import os
+import pathlib
+from .models import AbstractClassifier
+
+#!git clone https://github.com/DTrimarchi10/confusion_matrix.git
+import numpy as np
+from xgboost import XGBClassifier
+import xgboost as xgb
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import accuracy_score, confusion_matrix, matthews_corrcoef, mean_absolute_error
+from sklearn.metrics import f1_score, ConfusionMatrixDisplay
+#plot_confusion_matrix, is deprecated, so import ConfusionMatrixDisplay instead
+from sklearn.metrics import roc_curve, auc, precision_recall_curve, classification_report, average_precision_score
+import pandas as pd
+
+import seaborn as sns
+import matplotlib.pyplot as plt
+
+import os
+import sys
+import copy
+import json
+
+    
+
+class XGB(AbstractClassifier):
+    """Baseline model with a statically-defined architecture"""
+    def __init__(
+        self,
+        **kwargs,
+    )
+
+
+#    def call(self, inputs, **kwargs):
+#        features_input = inputs.get("features")
+#        dtrain = xgb.DMatrix(X_train, label=y_train)
+#        dtest = xgb.DMatrix(X_test, label=y_test)
+#
+#        return features_input, dtrain, dtest
+#
+    def summary(self, **kwargs):
+
+        model = xgb.train(
```

Comment:
Remember to use `self.model` here, in addition to any other places where you'd like to set an attribute (variable accessible by all the different methods in the `XGB` class).